BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BluetoothBase.java

```diff
@@ -35,53 +31,58 @@ public BluetoothBase(Context Context, BluetoothStatusChanged handler) {
 
         //bluetooth = new BluetoothAdapter(this);
         bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null) {
+        if (bluetooth != null) {
```

Comment:
If bluetooth is null then the class instance is useless. Please test and throw. This will, btw, let you remove all the tests for bluetooth == null everywhere else in this class since you know if the object exists then then bluetooth isn't null.